tokio-util/tests/task_join_queue.rs

```diff
@@ -192,23 +192,119 @@ async fn test_join_queue_join_next_with_id() {
 
     let (send, recv) = tokio::sync::watch::channel(());
 
-    let mut set = JoinQueue::new();
+    let mut queue = JoinQueue::new();
+    let mut spawned = Vec::with_capacity(TASK_NUM as usize);
+
+    for _ in 0..TASK_NUM {
+        let mut recv = recv.clone();
+        let handle = queue.spawn(async move { recv.changed().await.unwrap() });
+
+        spawned.push(handle.id());
+    }
+    drop(recv);
+
+    send.send_replace(());
+    send.closed().await;
+
+    let mut count = 0;
+    let mut joined = Vec::with_capacity(TASK_NUM as usize);
+    while let Some(res) = queue.join_next_with_id().await {
+        match res {
+            Ok((id, ())) => {
+                count += 1;
+                joined.push(id);
+            }
+            Err(err) => panic!("failed: {err}"),
+        }
+    }
+
+    assert_eq!(count, TASK_NUM);
+    assert_eq!(joined, spawned);
+}
+
+#[tokio::test]
+async fn test_join_queue_try_join_next() {
```

Comment:
Could you also test the to ensure the `try_join_next` has disabled the cooperative scheduling?